pkg/acsengine/defaults.go

```diff
@@ -646,14 +647,15 @@ func setAgentProfileDefaults(a *api.Properties, isUpgrade, isScale bool) {
 			profile.AcceleratedNetworkingEnabled = helpers.PointerToBool(!isUpgrade && !isScale && helpers.AcceleratedNetworkingSupported(profile.VMSize))
 		}
 
-		// don't default Distro for OpenShift
-		if !a.OrchestratorProfile.IsOpenShift() {
-			if profile.Distro == "" {
+		if profile.Distro == "" {
```

Comment:
```
if profile.Distro == "" {
			if a.OrchestratorProfile.IsKubernetes() && profile.OSDiskSizeGB >= api.VHDDiskSizeAKS {
				profile.Distro = api.AKS
			} else {
				profile.Distro = api.Ubuntu
			}
		}
```

Reply:
I've optimized it to that. Does it look okay?

Reply:
I have a slight preference for the current implementation as it emphasizes the k8s-specific conditions (which could grow over time based on more k8s-specific distros.

Reply:
The condition `profile.OSDiskSizeGB >= api.VHDDiskSizeAKS` is not equivalent to `profile.OSDiskSizeGB != 0 && profile.OSDiskSizeGB < api.VHDDiskSizeAKS`

Reply:
I see. profile.OSDiskSizeGB=0 represents a nil state I am assuming?

Reply:
Also again we don't want to set the distro for openshift

Reply:
Also what @CecileRobertMichon said. It's bit clumsy, but we want to include the zero value (0) when evaluating what to do. zero value means AKS distro, in this case.